assets/js/base/components/payment-methods/payment-method-options.js

```diff
@@ -42,52 +46,109 @@ const PaymentMethodOptions = () => {
 
-	return expressPaymentMethodActive ? null : (
-		<Tabs
-			className="wc-block-components-checkout-payment-methods"
-			onSelect={ ( tabName ) => {
-				setActivePaymentMethod( tabName );
-				removeNotice( 'wc-payment-error', noticeContexts.PAYMENTS );
-			} }
-			tabs={ Object.keys( paymentMethods ).map( ( name ) => {
-				const {
-					ariaLabel,
-					edit,
-					content,
-					label,
-					supports,
-				} = paymentMethods[ name ];
-				const component = isEditor ? edit : content;
-				return {
-					name,
-					title:
-						typeof label === 'string'
-							? label
-							: cloneElement( label, {
-									components:
-										paymentMethodInterface.components,
-							  } ),
-					ariaLabel,
-					content: (
-						<PaymentMethodTab
-							allowsSaving={ supports.savePaymentInfo }
-						>
-							{ cloneElement( component, {
-								activePaymentMethod,
-								...paymentMethodInterface,
-							} ) }
-						</PaymentMethodTab>
-					),
-				};
-			} ) }
-			initialTabName={ activePaymentMethod }
-			ariaLabel={ __(
-				'Payment Methods',
-				'woo-gutenberg-products-block'
-			) }
-			id="wc-block-payment-methods"
-		/>
-	);
+	const options = Object.keys( paymentMethods ).map( ( name ) => {
+		const { edit, content, label, supports } = paymentMethods[ name ];
+		const component = isEditor ? edit : content;
+		return {
+			value: name,
+			label:
+				typeof label === 'string'
+					? label
+					: cloneElement( label, {
+							components: paymentMethodInterface.components,
+					  } ),
+			name: `wc-saved-payment-method-token-${ name }`,
+			content: (
+				<PaymentMethodTab allowsSaving={ supports.savePaymentInfo }>
+					{ cloneElement( component, {
+						activePaymentMethod,
+						...paymentMethodInterface,
+					} ) }
+				</PaymentMethodTab>
+			),
+		};
+	} );
+
+	const updateToken = ( value ) => {
+		setActivePaymentMethod( value );
+		setActiveSavedToken( '' );
+		removeNotice( 'wc-payment-error', noticeContexts.PAYMENTS );
+	};
+
+	const customerHasSavedCards =
+		Object.keys( customerPaymentMethods ).length > 0;
+	let paymentMethodsSelectorUI = null;
+	if ( options.length > 2 || customerHasSavedCards ) {
+		paymentMethodsSelectorUI = (
+			<RadioControlAccordion
+				id={ 'wc-payment-method-options' }
+				selected={ activeSavedToken ? null : activePaymentMethod }
+				onChange={ updateToken }
+				options={ options }
+			/>
+		);
+	} else if ( options.length === 1 ) {
+		const option = options[ 0 ];
+		paymentMethodsSelectorUI = (
+			<div className="wc-block-components-checkout-payment-methods__single-option">
+				<div className="wc-block-components-checkout-payment-methods__single-option-content-label">
+					{ option.label }
+				</div>
+				<div className="wc-block-components-checkout-payment-methods__single-option-content">
+					{ option.content }
+				</div>
+			</div>
+		);
+	} else {
+		paymentMethodsSelectorUI = (
+			<Tabs
+				className="wc-block-components-checkout-payment-methods"
```

Comment:
Should this class be tweaked now that it's not the only payment methods component, e.g. `wc-block-components-checkout-payment-methods__two-options-tabs`. Not a big deal but seems appropriate since we now have three variations here.

Reply:
My main review comment got lost somehow .. rewrote it below (😭)

This is looking great. I think it should have one more review from someone else before merge (since we both worked on it).

I found a couple of styling issues. Functionality working well :)

- In two-tab mode, there's no box outline (figma has a box around the whole component, consistent with other modes).
- If `Order note` is disabled, the line overlaps the `Submit order` button.

<img width="781" alt="Screen Shot 2020-12-07 at 4 15 51 PM" src="https://user-images.githubusercontent.com/4167300/101307735-063b9d80-38ad-11eb-9325-d2e538ca47c1.png">

Probably a good idea to add all the block options that might be impacted to the test instructions – e.g. `Allow shopper to optionally add order notes`, `Show a "Return to cart" link` etc.

Reply:
The latest trunk was merged back into this branch.

I thought about renaming the `wc-block-components-checkout-payment-methods` but I can't convince myself to this `wc-block-components-checkout-payment-methods__two-options-tabs.` The logic used for displaying payments using one of the 3 options is complicated and finding naming that would make sense especially for the radio buttons case would not be possible. So I will leave the single as the special cease and not change the others. I will revisit this decision if the reviewer will mark this.

Outline added:
![image](https://user-images.githubusercontent.com/17271089/102088728-9896f080-3e1b-11eb-8a68-36dddad14612.png)

The line issue is correceted:
![image](https://user-images.githubusercontent.com/17271089/102088692-903eb580-3e1b-11eb-84b2-53b4b9fbc5da.png)

Reply:
> I thought about renaming the wc-block-components-checkout-payment-methods but I can't convince myself to this wc-block-components-checkout-payment-methods__two-options-tabs. 

Up to you - I don't think it's a blocker, will be interested to see what reviewers think. Naming it `__radio-buttons` is an option :)

The radiobuttons UI feels like the main/default option to me, the others feel like special cases/variations.